File: app/src/main/java/com/silentgames/silent_planet/customListeners/CustomScaleGestureListener.java
package com.silentgames.silent_planet.customListeners;

import android.view.ActionMode;
import android.view.ScaleGestureDetector;

/**
 * Created by gidroshvandel on 21.06.17.
 */
public class CustomScaleGestureListener extends ScaleGestureDetector.SimpleOnScaleGestureListener {

    private Callback callback;

    public interface Callback {

        void onScale();

    }

    public CustomScaleGestureListener(Callback callback) {
        this.callback = callback;
    }

    public boolean onScale(ScaleGestureDetector scaleGestureDetector) {

        callback.onScale();

        return true;
    }
}

File: android/src/com/silentgames/silent_planet/customListeners/CustomScaleGestureListener.kt
package com.silentgames.silent_planet.customListeners

import android.view.ScaleGestureDetector
import android.view.ScaleGestureDetector.SimpleOnScaleGestureListener

/**
 * Created by gidroshvandel on 21.06.17.
 */
class CustomScaleGestureListener(private val callback: Callback) : SimpleOnScaleGestureListener() {

    interface Callback {
        fun onScale()
    }

    override fun onScale(scaleGestureDetector: ScaleGestureDetector): Boolean {
        callback.onScale()
        return true
    }

}